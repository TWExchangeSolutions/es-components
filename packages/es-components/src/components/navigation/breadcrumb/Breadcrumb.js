import React from 'react';
import PropTypes from 'prop-types';
import styled, { ThemeProvider, withTheme } from 'styled-components';
import tinycolor from 'tinycolor2';
import genId from '../../util/generateAlphaName';
import defaultTheme from '../../theme/defaultTheme';

const OrderedList = styled.ol`
  background-color: ${props => props.theme.colors.grayLightest};
  border-radius: 2px;
  line-height: 1.4;
  list-style: none;
  margin-bottom: 25px;
  margin-top: 0;
  padding: 8px 15px;

  > li + li {
    padding-left: 24px;

    &::before {
      border-color: ${props => props.theme.colors.gray};
      border-style: solid;
      border-width: 0 2px 2px 0;
      box-sizing: border-box;
      content: '';
      display: block;
      height: 13px;
      left: 0px;
      position: absolute;
      top: 20%;
      transform: rotate(-45deg);
      width: 13px;
    }
  }
`;

const Crumb = styled.li`
  color: ${props => props.theme.colors.info};
  display: inline-block;
  margin-right: 2px;
  position: relative;

  > a {
    color: ${props => props.theme.colors.primary};
    text-decoration: none;
  }

  > a:hover {
    color: ${props =>
    tinycolor(props.theme.colors.primary)
      .darken(15)
      .toRgbString()};
    text-decoration: underline;
  }
`;

function Breadcrumb({ children, ...props }) {
  return (
    <ThemeProvider theme={props.theme}>
      <OrderedList {...props}>
        {React.Children.map(children, child => (
          <Crumb key={genId()}>{child}</Crumb>
        ))}
      </OrderedList>
    </ThemeProvider>
  );
}

Breadcrumb.propTypes = {
  children: PropTypes.node.isRequired,
  /**
   * Theme object used by the ThemeProvider,
   * automatically passed by any parent component using a ThemeProvider
   */
  theme: PropTypes.object
};

Breadcrumb.defaultProps = {
  theme: defaultTheme
};

export default withTheme(Breadcrumb);
