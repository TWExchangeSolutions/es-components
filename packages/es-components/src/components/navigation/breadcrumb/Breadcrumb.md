Use `Breadcrumb` to chain a list of items together. The last element, if not a
link, will be styled differently to denote the user's current location.
```
<Breadcrumb className="test">
  <a href="#home">Home</a>
  <a href="#gettingthere">On Our Way</a>
  <a href="#test">Getting There</a>
  <span>Destination</span>
</Breadcrumb>
```
