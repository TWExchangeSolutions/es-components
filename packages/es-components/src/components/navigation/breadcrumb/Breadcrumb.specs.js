/* eslint-env jest */

import React from 'react';
import { renderWithTheme } from 'styled-enzyme';
import Breadcrumb from './Breadcrumb';

describe('BreadcrumbTestSuite', () => {
  let instanceToRender;

  beforeEach(() => {
    instanceToRender = (
      <Breadcrumb className="test">
        <a href="/test">test</a>
        <a href="/test2">test2</a>
      </Breadcrumb>
    );
  });

  it('renders as expected', () => {
    const tree = renderWithTheme(instanceToRender).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
