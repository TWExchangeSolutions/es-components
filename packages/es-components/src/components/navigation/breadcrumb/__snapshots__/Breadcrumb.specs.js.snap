// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BreadcrumbTestSuite renders as expected 1`] = `
<ol
  className="test sc-bdVaJa gcwjiK"
>
  <li
    className="sc-bwzfXH jLUeGH"
  >
    <a
      href="/test"
    >
      test
    </a>
  </li>
  <li
    className="sc-bwzfXH jLUeGH"
  >
    <a
      href="/test2"
    >
      test2
    </a>
  </li>
</ol>
`;
